### PR TITLE
Json tabbed code sample

### PR DESCRIPTION
**Pull request by andy-stark-redis**
Code samples for the two JSON pages:
https://redis.io/docs/latest/develop/data-types/json/
https://redis.io/docs/latest/develop/data-types/json/path/

### CODE REVIEW

**Comment by sazzad16** [comment]
@bsbodden Would it be a sane ask to split the long Strings?

**Comment by bsbodden** [comment]
> @bsbodden Would it be a sane ask to split the long Strings?

Agreed, good idea

File: src/test/java/io/redis/examples/JsonExample.java [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+        Assert.assertEquals("[\"Phoebe\",\"Quaoar\"]", res34.toString());
+        // REMOVE_END
+
+
+        // STEP_START filter1
+        Object res35 = jedis.jsonGet(
+            "bikes:inventory",
+            new Path2(
+                "$..mountain_bikes[?(@.price < 3000 && @.specs.weight < 10)]"
+            )
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:392]:
```suggestion
            new Path2("$..mountain_bikes[?(@.price < 3000 && @.specs.weight < 10)]")
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+            + "\"Quaoar\",\"id\":\"bike:2\",\"colors\":[\"black\",\"white\"]}]",
+            res35.toString()
+        );
+        // REMOVE_END
+
+
+        // STEP_START filter2
+        Object res36 = jedis.jsonGet(
+            "bikes:inventory",
+            new Path2("$..[?(@.specs.material == 'alloy')].model")
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:416]:
```suggestion
            "bikes:inventory", new Path2("$..[?(@.specs.material == 'alloy')].model")
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+        // Tests for 'filter2' step.
+        // REMOVE_START
+        Assert.assertEquals("[\"Weywot\",\"Mimas\"]", res36.toString());
+        // REMOVE_END
+
+
+        // STEP_START filter3
+        Object res37 = jedis.jsonGet(
+            "bikes:inventory",
+            new Path2("$..[?(@.specs.material =~ '(?i)al')].model")
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:430]:
```suggestion
            "bikes:inventory", new Path2("$..[?(@.specs.material =~ '(?i)al')].model")
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+        // Tests for 'filter3' step.
+        // REMOVE_START
+        Assert.assertEquals("[\"Quaoar\",\"Weywot\",\"Salacia\",\"Mimas\"]", res37.toString());
+        // REMOVE_END
+
+
+        // STEP_START filter4
+        jedis.jsonSet(
+            "bikes:inventory",
+            new Path2("$.inventory.mountain_bikes[0].regex_pat"),
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:445]:
```suggestion
            "bikes:inventory", new Path2("$.inventory.mountain_bikes[0].regex_pat"),
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 5]
```diff
@@ -0,0 +1,547 @@
...
+            "\"(?i)al\""
+        );
+        jedis.jsonSet(
+            "bikes:inventory",
+            new Path2("$.inventory.mountain_bikes[1].regex_pat"),
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:450]:
```suggestion
            "bikes:inventory", new Path2("$.inventory.mountain_bikes[1].regex_pat"),
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 6]
```diff
@@ -0,0 +1,547 @@
...
+            "\"(?i)al\""
+        );
+        jedis.jsonSet(
+            "bikes:inventory",
+            new Path2("$.inventory.mountain_bikes[2].regex_pat"),
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:455]:
```suggestion
            "bikes:inventory", new Path2("$.inventory.mountain_bikes[2].regex_pat"),
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 7]
```diff
@@ -0,0 +1,547 @@
...
+        // Tests for 'filter4' step.
+        // REMOVE_START
+        Assert.assertEquals("[\"Quaoar\",\"Weywot\"]", res38.toString());
+        // REMOVE_END
+
+
+        // STEP_START update_bikes
+        Object res39 = jedis.jsonGet(
+            "bikes:inventory", new Path2("$..price")
+        );
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:477]:
```suggestion
        Object res39 = jedis.jsonGet("bikes:inventory", new Path2("$..price"));
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 8]
```diff
@@ -0,0 +1,547 @@
...
+        System.out.println(res39);
+        // >>> [1920,2072,3264,1475,3941]
+
+        Object res40 = jedis.jsonNumIncrBy(
+            "bikes:inventory", new Path2("$..price"), -100
+        );
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:483]:
```suggestion
        Object res40 = jedis.jsonNumIncrBy("bikes:inventory", new Path2("$..price"), -100);
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 9]
```diff
@@ -0,0 +1,547 @@
...
+        System.out.println(res40);  // >>> [1820,1972,3164,1375,3841]
+
+        Object res41 = jedis.jsonNumIncrBy(
+            "bikes:inventory", new Path2("$..price"), 100
+        );
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:488]:
```suggestion
        Object res41 = jedis.jsonNumIncrBy("bikes:inventory", new Path2("$..price"), 100);
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 10]
```diff
@@ -0,0 +1,547 @@
...
+        Assert.assertEquals("[1920,2072,3264,1475,3941]", res41.toString());
+        // REMOVE_END
+
+
+        // STEP_START update_filters1
+        jedis.jsonSet(
+            "bikes:inventory",
+            new Path2("$.inventory.*[?(@.price<2000)].price"),
+            1500
+        );
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:505]:
```suggestion
        jedis.jsonSet("bikes:inventory", new Path2("$.inventory.*[?(@.price<2000)].price"), 1500);
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 11]
```diff
@@ -0,0 +1,547 @@
...
+        Object res42 = jedis.jsonGet(
+            "bikes:inventory", new Path2("$..price")
+        );
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:508]:
```suggestion
        Object res42 = jedis.jsonGet("bikes:inventory", new Path2("$..price"));
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 12]
```diff
@@ -0,0 +1,547 @@
...
+        // Tests for 'update_filters1' step.
+        // REMOVE_START
+        Assert.assertEquals("[1500,2072,3264,1500,3941]", res42.toString());
+        // REMOVE_END
+
+
+        // STEP_START update_filters2
+        List<Long> res43 = jedis.jsonArrAppendWithEscape(
+            "bikes:inventory",
+            new Path2("$.inventory.*[?(@.price<2000)].colors"),
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:521]:
```suggestion
            "bikes:inventory", new Path2("$.inventory.*[?(@.price<2000)].colors"),
```

File: src/test/java/io/redis/examples/JsonExample.java [hunk 13]
```diff
@@ -0,0 +1,547 @@
...
+            "\"pink\""
+        );
+        System.out.println(res43);  // >>> [3, 3]
+
+        Object res44 = jedis.jsonGet(
+            "bikes:inventory", new Path2("$..[*].colors")
+        );
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:528]:
```suggestion
        Object res44 = jedis.jsonGet("bikes:inventory", new Path2("$..[*].colors"));
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/test/java/io/redis/examples/JsonExample.java [hunk 14]
```diff
@@ -0,0 +1,510 @@
...
+
+        List<Long> res17 = jedis.jsonArrAppendWithEscape("riders", new Path2("$"), "Norem");
+        System.out.println(res17);  // >>> [1]
+
+        Object res18 = jedis.jsonGet("riders", new Path2("$"));
+        System.out.println(res18);  // >>> [["Norem"]]
+
+        List<Long> res19 = jedis.jsonArrInsertWithEscape(
+            "riders", new Path2("$"), 1,
+            "Prickett", "Royce", "Castilla"
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/JsonExample.java:137]:
```suggestion
            "riders", new Path2("$"), 1, "Prickett", "Royce", "Castilla"
```

**Pull request merged**